Storage/Storage/CoreData/CoreDataManager.swift

```diff
@@ -120,10 +122,50 @@ public class CoreDataManager: StorageManagerType {
                 fatalError("☠️ [CoreDataManager] Unable to regenerate Persistent Store! \(error)")
             }
 
-            NSLog("💣 [CoreDataManager] Stack Destroyed!")
+            DDLogVerbose("💣 [CoreDataManager] Stack Destroyed!")
             NotificationCenter.default.post(name: .StorageManagerDidResetStorage, object: self)
         }
     }
+
+    private func migrateDataModelIfNecessary() {
+        guard FileManager.default.fileExists(atPath: storeURL.path) else {
+            DDLogInfo("No store exists at URL \(storeURL).  Skipping migration.")
+            return
+        }
+
+        guard let metadata = try? NSPersistentStoreCoordinator.metadataForPersistentStore(ofType: NSSQLiteStoreType, at: storeURL, options: nil) else {
+            return
+        }
+
+        guard managedModel.isConfiguration(withName: nil, compatibleWithStoreMetadata: metadata) == false else {
+            // Configuration is compatible, no migration necessary.
+            return
+        }
+
+        DDLogWarn("⚠️ [CoreDataManager] Migration required for persistent store")
+
+        // Extract model names
+        let versionPath = modelURL.appendingPathComponent("VersionInfo.plist").path
```

Comment:
Nitpicky - one of the standards we follow is to find all the strings and add them to a `Constants` struct at the end of the file so they are easier to find and edit down the road. So this could be `modelURL.appendingPathComponent(Constants.versionInfo).path`.